mesonbuild/wrap/wrap.py

```diff
@@ -294,10 +295,11 @@ def resolve(self, packagename: str, method: str, current_subproject: str = '') -
                 # Write a dummy wrap file in main project that redirect to the
                 # wrap we picked.
                 with open(main_fname, 'w') as f:
+                    filename = PurePath(os.path.relpath(self.wrap.filename, self.subdir_root)).as_posix()
```

Comment:
While using pathlib, you can also replace os.path.relpath() by Path().relative_to().